include/exadg/incompressible_navier_stokes/spatial_discretization/turbulence_model.cpp

```diff
@@ -243,61 +342,66 @@ TurbulenceModel<dim, Number>::calculate_filter_width(dealii::Mapping<dim> const
       break;
     case TurbulenceEddyViscosityModel::WALE:
-      wale_model(filter_width, velocity_gradient, model_constant, viscosity);
+      wale_turbulence_model(
+        filter_width, velocity_gradient, shear_rate_squared, model_constant, viscosity);
       break;
     case TurbulenceEddyViscosityModel::Sigma:
-      sigma_model(filter_width, velocity_gradient, model_constant, viscosity);
+      sigma_turbulence_model(filter_width, symmetric_velocity_gradient, model_constant, viscosity);
       break;
+    default:
+      AssertThrow(
+        false,
+        dealii::ExcMessage(
+          "This TurbulenceEddyViscosityModel is not implemented in viscosity_model.cpp ."));
   }
 }
 
 template<int dim, typename Number>
 void
-TurbulenceModel<dim, Number>::smagorinsky_model(scalar const & filter_width,
-                                                tensor const & velocity_gradient,
-                                                double const & C,
-                                                scalar &       viscosity) const
+ViscosityModel<dim, Number>::smagorinsky_turbulence_model(scalar const & filter_width,
+                                                          scalar const & shear_rate_squared,
+                                                          double const & C,
+                                                          scalar &       viscosity) const
 {
-  tensor symmetric_gradient =
-    dealii::make_vectorized_array<Number>(0.5) * (velocity_gradient + transpose(velocity_gradient));
-
-  scalar rate_of_strain = 2.0 * scalar_product(symmetric_gradient, symmetric_gradient);
-  rate_of_strain        = std::exp(0.5 * std::log(rate_of_strain));
-
   scalar factor = C * filter_width;
-
-  viscosity += factor * factor * rate_of_strain;
+  factor *= factor;
+  factor *= std::sqrt(shear_rate_squared);
+  viscosity += factor;
 }
 
 template<int dim, typename Number>
 void
-TurbulenceModel<dim, Number>::vreman_model(scalar const & filter_width,
-                                           tensor const & velocity_gradient,
-                                           double const & C,
-                                           scalar &       viscosity) const
+ViscosityModel<dim, Number>::vreman_turbulence_model(scalar const & filter_width,
+                                                     tensor const & velocity_gradient,
+                                                     tensor const & symmetric_velocity_gradient,
+                                                     double const & C,
+                                                     scalar &       viscosity) const
 {
   scalar       velocity_gradient_norm_square = scalar_product(velocity_gradient, velocity_gradient);
   Number const tolerance                     = 1.0e-12;
 
-  tensor tensor = velocity_gradient * transpose(velocity_gradient);
+  tensor tensor = dealii::make_vectorized_array<Number>(2.0) * symmetric_velocity_gradient;
```

Comment:
this seems to be an invalid change! Addition vs. multiplication. The same comments apply here as for the smagorinsky model.

Reply:
That is true! :( Great that you caught that! Ignoring this partly motivated a change of the interface to save computations (n case one wants turbulence + gen. Newtonian) - now, this could be useless. I will check once I did the adaptations, if passing the 'shear_rate_squared' is actually better than passing the velocity gradient alone.